### ISSUE TITLE
Android back button behavior in a modal on Xamarin Forms

### ISSUE THREAD

**Issue by patrick11994** [MvvmCross/MvvmCross#3914]
## 🐛 Bug Report

When navigating to a second page in a modal on Xamarin Forms, the device's (hardware) backbutton behaves differently than the backbutton on the top left of the screen and when compared to iOS and UWP.

### Expected behavior
The device's backbutton on Android should behave the same as the back button on the top left on the screen in a modal, navigating up the navigation stack in stead of closing the modal.

**Comparison:**
| Platform | Going back with... | Result |
|-----------|--------------------|--------|
| UWP | Back button on the top left in the window's title bar | ✅ Goes back to the previous screen inside modal |
| iOS | Back button on the top left in title bar | ✅ Goes back to the previous screen inside modal |
| Android | Back button provided by the app (top left) | ✅ Goes back to the previous screen inside modal |
| Android | Back button provided by the system (hardware/software back key or system-wide gestures) | ❌ Closes modal |

### Reproduction steps
1. Open a modal
2. Navigate to a second screen inside this modal
3. Press the device's (hardware) back key.
Step 3 will close the modal in stead of going back to the first screen in the modal.

I have made a [sample project](https://github.com/patrick11994/MvxXFModalNavigation) to demonstrate this.

### Configuration

**Version:** 7.x

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [x] :monkey: Xamarin.Forms

**Comment by Cheesebaron** [comment]
Probably related to this code: https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross.Forms/Platforms/Android/Views/MvxFormsApplicationActivity.cs#L177

**Comment by patrick11994** [comment]
I've performed some tests, and it looks like it is in here: https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross.Forms/Platforms/Android/Views/MvxFormsAppCompatActivity.cs#L178-L208

Funny is, when just remove all the lines, everything works as expected. I've tried to go through git history to find out why this is in there, but didn't find more information about it (no PR, no explanation). This is added >3 years ago. It may be unnecessary today?

Do I miss something? If not, I would like to create a PR for this 🙂
